staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -61,6 +61,13 @@ const (
 type ProfileApplier interface {
 	// Apply applies the profile to the given container in the pod.
 	Apply(pod *corev1.Pod, containerName string, target runtime.Object) error
+	NewKeepLabels()
```

Comment:
Instead of extending the interface just for this, I was thinking something similar;

```diff
diff --git a/staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go b/staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go
index 2e0f8621ee7..863fc07bd78 100644
--- a/staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go
+++ b/staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go
@@ -58,13 +58,64 @@ const (
        ProfileSysadmin = "sysadmin"
 )
 
+type KeepFields struct {
+       labels         bool
+       annotations    bool
+       liveness       bool
+       readiness      bool
+       startup        bool
+       initContainers bool
+}
+
+func NewKeepFields() KeepFields {
+       return KeepFields{
+               initContainers: true,
+       }
+}
+
+// removeLabels removes labels from the pod.
+func (keeps KeepFields) removeLabels(p *corev1.Pod) {
+       if !keeps.labels {
+               p.Labels = nil
+       }
+}
+
+// removeAnnotations remove annotations from the pod.
+func (keeps KeepFields) removeAnnotations(p *corev1.Pod) {
+       if !keeps.annotations {
+               p.Annotations = nil
+       }
+}
+
+// removeProbes remove probes from all containers of the pod.
+func (keeps KeepFields) removeProbes(p *corev1.Pod) {
+       for i := range p.Spec.Containers {
+               if !keeps.liveness {
+                       p.Spec.Containers[i].LivenessProbe = nil
+               }
+               if !keeps.readiness {
+                       p.Spec.Containers[i].ReadinessProbe = nil
+               }
+               if !keeps.startup {
+                       p.Spec.Containers[i].StartupProbe = nil
+               }
+       }
+}
+
+// removeInitContainers remove initContainers from the pod.
+func (keeps KeepFields) removeInitContainers(p *corev1.Pod) {
+       if !keeps.initContainers {
+               p.Spec.InitContainers = nil
+       }
+}
+
 type ProfileApplier interface {
        // Apply applies the profile to the given container in the pod.
        Apply(pod *corev1.Pod, containerName string, target runtime.Object) error
 }
 
 // NewProfileApplier returns a new Options for the given profile name.
-func NewProfileApplier(profile string) (ProfileApplier, error) {
+func NewProfileApplier(profile string, keepFields KeepFields) (ProfileApplier, error) {
        switch profile {
        case ProfileLegacy:
                return &legacyProfile{}, nil
@@ -87,12 +138,15 @@ type legacyProfile struct {
 }
 
 type generalProfile struct {
+       KeepFields
 }
 
 type baselineProfile struct {
+       KeepFields
 }
 
 type restrictedProfile struct {
+       KeepFields
 }
 
 type netadminProfile struct {
@@ -164,7 +218,7 @@ func (p *baselineProfile) Apply(pod *corev1.Pod, containerName string, target ru
 
        switch style {
        case podCopy:
-               removeLabelsAndProbes(pod)
+               p.KeepFields.removeAnnotations(pod)
                shareProcessNamespace(pod)
 
        case ephemeral, node:
@@ -242,17 +296,6 @@ func (p *sysadminProfile) Apply(pod *corev1.Pod, containerName string, target ru
        return nil
 }
 
-// removeLabelsAndProbes removes labels from the pod and remove probes
-// from all containers of the pod.
-func removeLabelsAndProbes(p *corev1.Pod) {
-       p.Labels = nil
-       for i := range p.Spec.Containers {
-               p.Spec.Containers[i].LivenessProbe = nil
-               p.Spec.Containers[i].ReadinessProbe = nil
-               p.Spec.Containers[i].StartupProbe = nil
-       }
-}
-

```